modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+    }
+};// GAPI_OCV_KERNEL(BuildFaces)
+
+GAPI_OCV_KERNEL(OCVRunNMS, RunNMS) {
+    static void run(const std::vector<Face> &in_faces,
+        double threshold,
+        bool useMin,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        out_faces = Face::runNMS(in_faces_copy, threshold, useMin);
+    }
+};// GAPI_OCV_KERNEL(RunNMS)
+
+GAPI_OCV_KERNEL(OCVAccumulatePyramidOutputs, AccumulatePyramidOutputs) {
+    static void run(const std::vector<Face> &total_faces,
+        const std::vector<Face> &in_faces,
+        std::vector<Face> &out_faces) {
+        out_faces = total_faces;
+        out_faces.insert(out_faces.end(), in_faces.begin(), in_faces.end());
+    }
+};// GAPI_OCV_KERNEL(AccumulatePyramidOutputs)
+
+GAPI_OCV_KERNEL(OCVApplyRegression, ApplyRegression) {
+    static void run(const std::vector<Face> &in_faces,
+        bool addOne,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        Face::applyRegression(in_faces_copy, addOne);
+        out_faces.clear();
+        out_faces.insert(out_faces.end(), in_faces_copy.begin(), in_faces_copy.end());
+    }
+};// GAPI_OCV_KERNEL(ApplyRegression)
+
+GAPI_OCV_KERNEL(OCVBBoxesToSquares, BBoxesToSquares) {
+    static void run(const std::vector<Face> &in_faces,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        Face::bboxes2Squares(in_faces_copy);
+        out_faces.clear();
+        out_faces.insert(out_faces.end(), in_faces_copy.begin(), in_faces_copy.end());
+    }
+};// GAPI_OCV_KERNEL(BBoxesToSquares)
+
+GAPI_OCV_KERNEL(OCVR_O_NetPreProcGetROIs, R_O_NetPreProcGetROIs) {
+    static void run(const std::vector<Face> &in_faces,
+        const cv::Size & in_image_size,
+        std::vector<cv::Rect> &outs) {
+        outs.clear();
+        for (auto& f : in_faces) {
+            cv::Rect tmp_rect = f.bbox.getRect();
+            //Compare to transposed sizes width<->height
+            tmp_rect &= cv::Rect(tmp_rect.x, tmp_rect.y, in_image_size.height - tmp_rect.x - 4, in_image_size.width - tmp_rect.y - 4);
+            outs.push_back(tmp_rect);
+        }
+    }
+};// GAPI_OCV_KERNEL(R_O_NetPreProcGetROIs)
+
+
+GAPI_OCV_KERNEL(OCVRNetPostProc, RNetPostProc) {
+    static void run(const std::vector<Face> &in_faces,
+        const std::vector<cv::Mat> &in_scores,
+        const std::vector<cv::Mat> &in_regresssions,
+        double threshold,
+        std::vector<Face> &out_faces) {
+        out_faces.clear();
+        for (unsigned int k = 0; k < in_faces.size(); ++k) {
+            const float* scores_data = (float*)in_scores[k].data;
+            const float* reg_data = (float*)in_regresssions[k].data;
+            if (scores_data[1] >= threshold) {
+                Face info = in_faces[k];
+                info.score = scores_data[1];
+                for (int i = 0; i < 4; ++i) {
+                    info.regression[i] = reg_data[i];
```

Comment:
Can fields of `Face` be a STL containers? This loop is equivalent to =.

Reply:
???

Reply:
I believe declaring `.regression` as `std::array<float,4>` will make the trick. Still, the RHS of the assignment is raw float* so maybe even `std::copy_n` works. :)

Reply:
Done